tests/test_signals.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tests for the ``django-user-tasks`` Celery signal handlers."""
+
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+
+from django.contrib.auth.models import User
+from django.test import override_settings, TestCase
+
+from celery import chain, chord, group, shared_task
+import pytest
+
+from user_tasks.models import UserTaskStatus
+from user_tasks.tasks import UserTask
+
+LOGGER = logging.getLogger(__name__)
+USER_ID = 1
+
+
+class SampleTask(UserTask):
+
+    @staticmethod
+    def generate_name(arguments_dict):
+        return 'SampleTask: {}'.format(arguments_dict['argument'])
+
+    @staticmethod
+    def calculate_total_steps(arguments_dict):
+        return arguments_dict['kwargs'].get('total_steps', UserTask.calculate_total_steps(arguments_dict))
+
+
+@shared_task(base=SampleTask, bind=True)
+def sample_task(self, user_id, argument, **kwargs):  # pylint: disable=unused-argument
+    print('Ran SampleTask for argument "{}"'.format(argument))
+    return argument
+
+
+@shared_task(base=SampleTask, bind=True)
+def missing_user_id(self, *args, **kwargs):  # pylint: disable=unused-argument
+    pass
+
+
+@shared_task(bind=True)
+def normal_task(self, *args, **kwargs):  # pylint: disable=unused-argument
+    return 'placeholder'
+
+
+class TestCreateUserTask(TestCase):
+    """
+    Tests of UserTaskStatus creation for new UserTasks.
+
+    Note that in Celery 3.1, sub-tasks of a group which are executed eagerly
+    (such as in many test suites) have no access to information about the
+    parent group.  Hence UserTaskStatus records cannot be automatically
+    created for the group as a whole in such tests (even though it works for
+    normal execution in worker processes).  This seems to be fixed in
+    Celery 4, which as of this writing has not yet been released.
+    """
```

Comment:
Nit: Substituting this text with commented-out test cases or lines in existing test cases that would fail might be more clear.